src/index.js

```diff
@@ -357,6 +375,54 @@ export default class AuthnWidget {
     }
   }
 
+  registerMfaDevicePairingEventHandler() {
+    Array.from(document.querySelectorAll('[data-mfa-device-pairing-selection]'))
+      .forEach(element => element.addEventListener('click', this.handleMfaDevicePairingSelection));
+  }
+
+  handleMfaDevicePairingSelection(evt) {
+    evt.preventDefault();
+    let source = evt.currentTarget;
+    if (source) {
+      let devicePairingMethod = source.dataset['mfaDevicePairingSelection'].split('.');
+      let data = {
+        'devicePairingMethod': {
+          'deviceType': devicePairingMethod[0]
+        }
+      };
+      if (devicePairingMethod.length > 1 && devicePairingMethod[1] !== '') {
+        data['devicePairingMethod']['applicationName'] = devicePairingMethod[1];
+      }
+      this.store.dispatch('POST_FLOW', "selectDevicePairingMethod", JSON.stringify(data));
+    } else {
+      console.log("ERROR - Unable to dispatch device selection as the target was null");
+    }
+  }
+
+  async postMobileActivationRequired() {
+    let data = this.store.getStore();
+    let QRCode = require('qrcode');
+    QRCode.toCanvas(document.getElementById('qrcode'), data.pairingKey, function (error) {
```

Comment:
Do we need to render the QR code each time this is called? Maybe the QR code can change between requests?

Reply:
Good call -- updated. It should be the same.